### PR TITLE
gives all xenos access to the (xeno) tacmap while the queen is on ovi

### PR DESCRIPTION
**Pull request by harryob**
title

:cl:
add: all xenos can now access the tacmap, while the queen is on ovi
/:cl:

### CODE REVIEW

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Seems to still work while queen is off ovi

File: code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm [hunk 1]
```diff
@@ -479,3 +479,60 @@
 	// Perform check_state(TRUE) silently:
 	if(xeno && !xeno.is_mob_incapacitated() || !xeno.buckled || !xeno.evolving && xeno.plasma_stored >= plasma_cost)
 		return TRUE
+
+/datum/action/xeno_action/onclick/tacmap
+	name = "View Tactical Map"
+	action_icon_state = "toggle_queen_zoom"
+	ability_name = "view tacmap"
+
+	var/mob/living/carbon/xenomorph/queen/tracked_queen
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm:488]: null this out on destroy

File: code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm [hunk 2]
```diff
@@ -479,3 +479,60 @@
...
+
+/datum/action/xeno_action/onclick/tacmap/give_to(mob/living/carbon/xenomorph/xeno)
+	. = ..()
+
+	RegisterSignal(xeno.hive, COMSIG_HIVE_NEW_QUEEN, PROC_REF(handle_new_queen))
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm:493]: is having a hive guaranteed in 100% of all cases?

File: code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm [hunk 3]
```diff
@@ -479,3 +479,60 @@
...
+	if(!xeno.hive.living_xeno_queen)
+		hide_from(xeno)
+		return
+
+	if(!xeno.hive.living_xeno_queen.ovipositor)
+		hide_from(xeno)
+
+	handle_new_queen(new_queen = xeno.hive.living_xeno_queen)
+
+/datum/action/xeno_action/onclick/tacmap/proc/handle_new_queen(datum/hive_status/hive, mob/living/carbon/xenomorph/queen/new_queen)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm:504]: codedoc pls

File: code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm [hunk 4]
```diff
@@ -479,3 +479,60 @@
...
+	tracked_queen = new_queen
+
+	if(!tracked_queen.ovipositor)
+		hide_from(owner)
+
+	RegisterSignal(tracked_queen, COMSIG_QUEEN_MOUNT_OVIPOSITOR, PROC_REF(handle_mount_ovipositor))
+	RegisterSignal(tracked_queen, COMSIG_QUEEN_DISMOUNT_OVIPOSITOR, PROC_REF(handle_dismount_ovipositor))
+	RegisterSignal(tracked_queen, COMSIG_PARENT_QDELETING, PROC_REF(handle_queen_qdel))
+
+/datum/action/xeno_action/onclick/tacmap/proc/handle_mount_ovipositor()
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm:519]: ditto for this and the other two procs

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm (hunk 2 above)
Comment (harryob) [reply code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm:497]: yep!

**Pull request merged**